readthedocs/projects/models.py

```diff
@@ -241,14 +241,28 @@ class Project(models.Model):
         blank=True,
         help_text=_('URL that documentation is expected to serve from'),
     )
+    # NOTE: single_version and single_language are mutually exclusive.
+    # single_version will have precedence over single_language.
+    # Maybe we should remove both, and have a new option field (versioning_type?).
```

Comment:
We definitely want to validate this and raise an error if someone sets both, but I agree it might be more sensible as a dropdown, but not 100% sure what's best.

Reply:
Created https://github.com/readthedocs/readthedocs.org/issues/10781 to talk more about this